### ISSUE TITLE
Form not updating when schema changes in component scope

### ISSUE THREAD

**Issue by mikebski** [mozilla-services/react-jsonschema-form#862]
### Prerequisites

- [x ] I have read the [documentation](https://github.com/mozilla-services/react-jsonschema-form/blob/master/README.md#readme);

### Description

Basically, when I have a form and dynamically change the schema due to an ajax request, the form does not re-render when the schema updates.

        <Form schema={cas}
                          uiSchema={createAuditUISchema}
                          onSubmit={(data) => {
                              create(token, data)
                          }}
                    />

In other words, the above does not update when I change `cas`

Here is a JSFiddle demonstrating the change to the schema not making it into the form: https://jsfiddle.net/mbaranski/pt20o4w7/

Here is the SO question:

https://stackoverflow.com/questions/49161710/react-react-jsonschema-form-not-re-rendering-when-schema-changes?noredirect=1#comment85329291_49161710

I would expect the form to update when the schema updates

**Comment by johncrisostomo** [comment]
I took a look at your JSFiddle and when I changed the `schema` prop to the `Form` to `this.state.schema` it seems to be working. Props is not being changed in that example because the assignment mutates the local state and not the props (as it should be).

I also looked at your SO question and seems to me that you need to check your reducer to make sure that the schema is really being updated (if it were me I would not put that value computation in `mapStateToProps`, but that's just my opinion).

**Comment by edi9999** [comment]
I also think this is an issue in your code (props used instead of state). I'm closing this but feel free to ping me if you still have the issue.

**Comment by arthur-mts** [comment]
The Form is rerendered when the formData is changed
So, i clean form data after change schema to fix this problem

**Comment by arthur-mts** [comment]
`const aditForm = (e) => {
    e.preventDefault();
    setSchema((oldSchema)=>{
      oldSchema.properties[`${name}`] = {type};
      return oldSchema;
    });
    setFormData({});
  };`

**Comment by arthur-mts** [comment]
`javascript
const aditForm = (e) => {
    e.preventDefault();
    setSchema((oldSchema)=>{
      oldSchema.properties[`${name}`] = {type};
      return oldSchema;
    });
    setFormData({});
  };
`

**Comment by arthur-mts** [comment]
```javascript
const aditForm = (e) => {
    e.preventDefault();
    setSchema((oldSchema)=>{
      oldSchema.properties[`${name}`] = {type};
      return oldSchema;
    });
    setFormData({});
  };
```
